log/handlers.ts

```diff
@@ -180,18 +178,30 @@ export class FileHandler extends WriterHandler {
   }
 
   log(msg: string) {
-    if (this._encoder.encode(msg).byteLength + 1 > this._buf.available()) {
+    const bytes = this._encoder.encode(msg + "\n");
+    if (bytes.byteLength > this._buf.byteLength - this._pointer) {
       this.flush();
     }
-    this._buf.writeSync(this._encoder.encode(msg + "\n"));
+    this._buf.set(bytes, this._pointer);
+    this._pointer += bytes.byteLength;
   }
 
   flush() {
-    if (this._buf?.buffered() > 0) {
-      this._buf.flush();
+    if (this._pointer > 0 && this._file) {
+      let written = 0;
+      while (written < this._pointer) {
+        written += this._file.writeSync(
+          this._buf.subarray(written, this._pointer),
+        );
+      }
+      this.resetBuffer();
     }
   }
 
+  resetBuffer() {
```

Comment:
Let's make this a private method. I don't think there's a need to introduce another public method to the API at this point.

Reply:
Done